R/opts.R

```diff
@@ -917,6 +917,299 @@ opt_table_font <- function(
+      summary_row.border.color = params$location_hlines_color,
+      grand_summary_row.border.color = params$location_hlines_color,
+      column_labels.background.color = params$column_labels_background_color,
+      stub.background.color = params$stub_background_color,
+      stub.border.style = params$stub_border_style,
+      stub.border.color = params$stub_border_color,
+      table_body.border.top.color = params$location_hlines_color,
+      table_body.border.bottom.color = params$location_hlines_color,
+      table_body.hlines.style = params$data_hlines_style,
+      table_body.hlines.color = params$data_hlines_color,
+      table_body.vlines.style = params$data_vlines_style,
+      table_body.vlines.color = params$data_vlines_color,
+      summary_row.background.color = params$summary_row_background_color,
+      grand_summary_row.background.color = params$grand_summary_row_background_color,
+      row.striping.background_color = params$row_striping_background_color
+    )
+
+  if (!is.na(params$table_outline_color)) {
+
+    tbl_colorized <-
+      opt_table_outline(
+        tbl_colorized,
+        width = px(3),
+        color = params$table_outline_color
+      )
+  }
+
+  tbl_colorized
+}
+
+# Colors for the `preset_colorized()` function
+preset_colors <-
+  list(
+    gray =    c("#F4F4F4", "#D5D5D5", "#929292", "#5F5F5F", "#000000"),
+    blue =    c("#EDF7FC", "#89D3FE", "#00A1FF", "#0076BA", "#004D80"),
+    orange =  c("#FFFFDB", "#FEFC66", "#FAE231", "#F8BA00", "#FF9400"),
+    aqua =    c("#EBFBF9", "#A5FEF2", "#14E6CF", "#01A99D", "#007C77"),
+    pink =    c("#FCF2F7", "#FF8CC6", "#EF5FA7", "#CB2A7B", "#99195F"),
+    green =   c("#EDF6E8", "#9AFD69", "#60D937", "#1EB100", "#027101"),
+    red =     c("#FEEDEC", "#FF968D", "#FF644E", "#ED220D", "#B51800")
+  )
+
+# Function for getting a vector of preset colors at a specified 'level'
+get_all_preset_colors_at_index <- function(index) {
+
+  preset_colors %>%
+    vapply(
+      FUN.VALUE = character(1),
+      USE.NAMES = FALSE,
+      FUN = function(x) x[[index]]
+    )
+}
+
+# Create a tibble that contains all style and color parameters for the
+# themed tables generated by the `preset_colorized()` function
+styles_colors_params <-
+  dplyr::tibble() %>%
+  dplyr::bind_rows(
+    dplyr::tibble( # style 1
+      color = names(preset_colors),
+      style = 1,
+      table_hlines_color = get_all_preset_colors_at_index(5),
+      location_hlines_color = get_all_preset_colors_at_index(4),
+      column_labels_background_color = "#FFFFFF",
+      stub_background_color = get_all_preset_colors_at_index(4),
+      stub_border_style = "solid",
+      stub_border_color = get_all_preset_colors_at_index(4),
+      data_hlines_style = "none",
+      data_hlines_color = get_all_preset_colors_at_index(2),
+      data_vlines_style = "none",
+      data_vlines_color = get_all_preset_colors_at_index(2),
+      summary_row_background_color = "#FFFFFF",
```

Comment:
I wonder if there's a design that plays somewhat better with CSS and other format-specific styling infra. (`#ffffff` background will instantly break with dark mode or any other background CSS setting..)

Reply:
Do you know of any testing tools to get a handle on this? Definitely interested in accommodating dark mode with the color choices.